plugins/discourse_webhooks.rb

```diff
@@ -0,0 +1,90 @@
+#
+# This cinch plugin is part of EV0002
+#
+# written by Ghabry <gabriel @ mastergk . de> 2018
+# available under ISC license
+#
+
+require "json"
+
+class Cinch::DiscourseWebhooks
+  include Cinch::Plugin
+  extend Cinch::HttpServer::Verbs
+
+  post "/discourse_webhook" do
+    request.body.rewind
+    payload = request.body.read
+
+    # check X-Discourse-Event-Signature, to ensure authorization
+    secret = bot.config.plugins.options[Cinch::DiscourseWebhooks][:secret]
+    signature = 'sha256=' + OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new('sha256'), secret, payload)
+    halt 403 unless Rack::Utils.secure_compare(signature, request.env['HTTP_X_DISCOURSE_EVENT_SIGNATURE'])
+
+    # return if we got an x-www-form-urlencoded request
+    halt 400 if params[:payload]
+
+    # return if we got no valid json data
+    data = JSON.parse(payload)
+    halt 400 if data.empty?
+
+    # get event type
+    event = request.env['HTTP_X_DISCOURSE_EVENT']
+
+    # handle event
+    case event
+    when "ping"
+
+      message = "Pong!"
+
+    when "post_created"
+      # return if we got no valid post data
+      post = data["post"]
+      halt 400 if post.nil?
+
+      template = 'New %s "%s" by %s (https://community.easyrpg.org/t/%s)'
```

Comment:
I would like to keep plugins generic, if possible. So we should add some `:url` key to options in `EV0002.rb`.